.github/workflows/migration-review.yml

```diff
@@ -0,0 +1,43 @@
+name: Migration Review
+on:
+  pull_request_target:
+    types: [opened]
+    paths:
+      - 'core/server/data/schema/**'
+      - 'core/server/data/migrations/versions/**'
+jobs:
+  createComment:
+    runs-on: ubuntu-latest
+    name: Create checklist comment
+    steps:
+      - uses: peter-evans/create-or-update-comment@v1
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            It looks like this PR contains a migration 👀
+            Here's the checklist for reviewing migrations:
+
+            ### General requirements
+
+            - [ ]  Satisfies idempotency requirement (both `up()` and `down()`)
+            - [ ]  Does not reference models
+            - [ ]  Filename is in the correct format
+            - [ ]  All code paths have appropriate log messages
+            - [ ]  Uses the correct utils
+            - [ ]  Contains a minimal changeset
+            - [ ]  Does not mix DDL/DML operations
+
+            ### Schema changes
+
+            - [ ]  Both schema change and related migration have been implemented
+            - [ ]  For column changes: indexes have been recreated for sqlite
+            - [ ]  For index changes: has been performance tested for large tables
+            - [ ]  For new tables/columns: fields use the appropriate predefined field lengths
+            - [ ]  For new tables/columns: field names follow the appropriate conventions
+            - [ ]  Does not drop a non-alpha table outside of a major version
+            - [ ]  For settings updates: follows the appropriate guidelines
+
+            ### Data changes
```

Comment:
Perhaps also something like "handles invalid data" (to cover JSON parsing errors) and "handles missing data" (checking for knex result length)?